### PR TITLE
Des 304 change of password different from the current one

### PR DESCRIPTION
**Pull request by cristianarce89**

### CODE REVIEW

**Review by Daniielmayo — APPROVED** [review]

File: components/organisms/ChangePasswordUser/ChangePasswordUser.tsx [hunk 1]
```diff
@@ -78,27 +46,20 @@ export const ChangePasswordUser = () => {
...
 
     try {
-      const responseUpdate = await fetch(
-        `${process.env.NEXT_PUBLIC_BACKEND_URL}/auth/change-password/${session?.user?.uid}`,
-        {
-          method: "PUT",
-          headers: {
-            "Content-Type": "application/json",
+      if (userInfo?.uid !== undefined) {
+        const changePasswordUser = await changePassword(
```
Comment (SthefaniaEcheverry) [inline components/organisms/ChangePasswordUser/ChangePasswordUser.tsx:50]: This variable should be `changeUserPassword`

File: components/organisms/ChangePasswordUser/ChangePasswordUser.tsx [hunk 2]
```diff
@@ -78,27 +46,20 @@ export const ChangePasswordUser = () => {
...
+          userInfo?.uid,
+          password
+        );
+        Swal.fire({
+          icon: "success",
+          title: "Contraseña modificada",
+          text: "Los cambios en tu perfil han sido guardados exitosamente",
+          didClose: () => {
```
Comment (SthefaniaEcheverry) [inline components/organisms/ChangePasswordUser/ChangePasswordUser.tsx:58]: The name of this attribute isn't clear, can you change it for any other more descriptive? pls  

It looks like a function to redirect after closing the modal, so I suggest something like `redirectAfterClose`

File: components/organisms/ChangePasswordUser/ChangePasswordUser.tsx [hunk 3]
```diff
@@ -107,14 +68,37 @@ export const ChangePasswordUser = () => {
       );
     }
   };
+  const validatePasswordUser = async () => {
```
Comment (SthefaniaEcheverry) [inline components/organisms/ChangePasswordUser/ChangePasswordUser.tsx:71]: This variable should be `validateUserPassword`

File: components/organisms/ChangePasswordUser/ChangePasswordUser.tsx [hunk 4]
```diff
@@ -107,14 +68,37 @@ export const ChangePasswordUser = () => {
...
+    if (userInfo?.uid !== undefined) {
+      const validatePasswordData = await validatePassword(
```
Comment (SthefaniaEcheverry) [inline components/organisms/ChangePasswordUser/ChangePasswordUser.tsx:73]: This variable should be `passwordValidation` or `isValidatedPassword`

File: components/organisms/ChangePasswordUser/ChangePasswordUser.tsx [hunk 5]
```diff
@@ -107,14 +68,37 @@ export const ChangePasswordUser = () => {
...
+          return true;
+        }
+      } else {
+        setErrorMessage(validatePasswordData.msg);
+        return false;
+      }
+    }
+  };
 
   const validatePasswordSubmit = async (
```
Comment (SthefaniaEcheverry) [inline components/organisms/ChangePasswordUser/ChangePasswordUser.tsx:95]: This function should be `validateSubmitPassword`

File: components/organisms/ChangePasswordUser/ChangePasswordUser.tsx [hunk 6]
```diff
@@ -107,14 +68,37 @@ export const ChangePasswordUser = () => {
...
     e: React.FormEvent<HTMLFormElement>
   ) => {
     e.preventDefault();
 
     try {
-      const isPasswordValid = await validatePassword();
+      const isPasswordValid = await validatePasswordUser();
```
Comment (SthefaniaEcheverry) [inline components/organisms/ChangePasswordUser/ChangePasswordUser.tsx:101]: This variable should be `isValidatedUserPassword`

File: services/user/changePassword.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
+export const changePassword = async (id: string, password: string) => {
+    try {
+        const responseData = await fetch(
```
Comment (SthefaniaEcheverry) [inline services/user/changePassword.ts:3]: This variable should be `dataResponse`

**Review by SthefaniaEcheverry — CHANGES_REQUESTED** [review]
I left you some comments

File: components/organisms/ChangePasswordUser/ChangePasswordUser.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (cristianarce89) [reply components/organisms/ChangePasswordUser/ChangePasswordUser.tsx:50]: change variable

File: components/organisms/ChangePasswordUser/ChangePasswordUser.tsx (hunk 2 above)
Comment (cristianarce89) [reply components/organisms/ChangePasswordUser/ChangePasswordUser.tsx:58]: included attribute of swil.fire and next

File: components/organisms/ChangePasswordUser/ChangePasswordUser.tsx (hunk 3 above)
Comment (cristianarce89) [reply components/organisms/ChangePasswordUser/ChangePasswordUser.tsx:71]: changed

File: components/organisms/ChangePasswordUser/ChangePasswordUser.tsx (hunk 4 above)
Comment (cristianarce89) [reply components/organisms/ChangePasswordUser/ChangePasswordUser.tsx:73]: changed This variable should be validateUserPassword

Comment (cristianarce89) [reply components/organisms/ChangePasswordUser/ChangePasswordUser.tsx:73]: changed passwordValidation

File: components/organisms/ChangePasswordUser/ChangePasswordUser.tsx (hunk 5 above)
Comment (cristianarce89) [reply components/organisms/ChangePasswordUser/ChangePasswordUser.tsx:95]: change validateSubmitPassword

File: components/organisms/ChangePasswordUser/ChangePasswordUser.tsx (hunk 6 above)
Comment (cristianarce89) [reply components/organisms/ChangePasswordUser/ChangePasswordUser.tsx:101]: changed validateUserPassword

File: services/user/changePassword.ts (hunk 7 above)
Comment (cristianarce89) [reply services/user/changePassword.ts:3]: changed

**Review by SthefaniaEcheverry — APPROVED** [review]

**Review by abdulflorez — APPROVED** [review]

**Pull request merged**